packages/core/core/src/BundleGraph.js

```diff
@@ -185,6 +185,29 @@ export default class BundleGraph {
     );
   }
 
+  isAssetReferencedByType(asset: Asset, type: string): boolean {
+    return this._graph
+      .getNodesConnectedTo(
+        nullthrows(this._graph.getNode(asset.id)),
+        'references'
+      )
+      .some(v => v.type === type);
+  }
+
+  hasParentBundleOfType(bundle: Bundle, type: string): boolean {
```

Comment:
There aws another issue in the function: it used`bundle.isEntry`, but it should really check: does `bundle` have any JS parent bundles (only the top level JS bundle should insert the prelude), not sure if this is the right way